fishes

jonkeda opened the pull request:
Hi,

This pull request contains the fishing-inserter from the mashmod and I also added 6 fish foods. From simple cooked fish to taste Fish and Chips.

I hope you like it.

Cheers,
Daan

Commit:
fishes

Commit:
fishes

Oceanel51 commented on control.lua:
```diff
@@ -342,9 +384,28 @@ script.on_event({defines.events.on_tick}, function (e)
+end
 
 script.on_event(defines.events.on_built_entity, function(event)
+	local_on_added(event)
```
After this line is not executed further code - when the farm is built it does not add fuel.

jonkeda replied:
hi oceanel51. That was a bug, I reconnected the event in the correct way and the wood was added.

But, that piece of code is also the bug that I reported. It's the reason that after a while the farm starts eating Compost like crazy. I removed the code and changed the fuel value 100kj. And that made the farms work more in a way that atleast I liked.

I hope you agree otherwise the code is commented so you can change that.

Oceanel51 commented:
Hi, thanks for Your work!
I checked a bit and found one mistake, marked her in review.
I recently started learning lua and do not quite understand what is not working properly there.

I'll never played modmash :(

Commit:
fishes

Commit:
fishes

jonkeda commented:
Added

an extra "Burner fishing robot"
Added and Apple which you get cutting down trees
Added a recipe that let's you eat apples and create Compost

Changed the localnames of some recipes: 
for instance From: Cucumber seeds To: Process cucumbers 
and from: Cucumber To: Harvest cucumber

Changed the first growing recipes to use half of the ingredients and also get half of the produce.

Changed the Earth to Compost recipe from 100 to 10. Seems more fun.

Changed the Hunger-speed minimumvalue to 1.

jonkeda commented:
Hi Oceanel51. Thanks for the comments. If you have time please review the new changes. I need the apples for the colonists in the starting period to eat. Until they get the first farms up and running.

If you want to play with the Colonists checkout: https://github.com/jonkeda/ColonialCharter. I can't upload the colonialindustry mod at the moment as it depends on the apple in Food Industry.

Mashmod is something I came across, it seemed fun to add the fishing machine to FoodIndustry. Mashmod also has Biters as Pets maybe we can use that idea to add Cattle to FoodIndustry.

I am just learning LUA and Factorio myself. But it seems fun at the moment.

Commit:
added oranges and a foodpicker. It's looks like the fisher robot. But put it next to a tree and it will start picking fruit.

jonkeda commented:
Hi I added Oranges and a food picker. Put it next to a tree and it will start picking apples or oranges, depending what is on the trees next to the picker. Picker range is 5. 

I am really bad at art so the apple and orange icons need to be made gray instead of white. And the color of the food picker could be different, green maybe?

todo:
- Rotation of the picker should be fixed also. It currently doesn't change when picking fruits from tree. Just eye candy but still. 
- It should only pick if it has power.
- Fruit should not hang on dead or dry trees. 

Hope you like it.

I like the trello board. Unfortunatelly I can't read the cyrillic characters. Where are you from? I am from the Netherlands.

Oceanel51 committed:
change some sprites

Oceanel51 commented:
> added Oranges and a food picker
> 
orange.png lost :)

> I am really bad at art so the apple and orange icons...
> 
This is not a problem i will fix. And in general, all that concerns the graphics is not difficult for me.

> * Rotation of the picker should be fixed also. It currently doesn't change when picking fruits from tree. Just eye candy but still.
> * It should only pick if it has power.
> * Fruit should not hang on dead or dry trees.
> 
> Hope you like it.
> 
Yes. But see this card https://trello.com/c/G4wF2X5q, I translated it for consideretion. It's my idea how you can use existing farms to grow trees.
In general, these should be separate types of trees, sprites i will do. ;)

>I can't read the cyrillic characters.
>
I did not have time to translate this yet. I don't speak English very well. I used to translate, but I lost my thoughts, which I wanted to write. So first write, then translate.

> Where are you from?
I'm from Ukraine.

Oceanel51 committed:
fixes

better apple
fix apple and orange capsules

jonkeda commented:
Your english is as good as mine. So no worries. I can also use google translate for the cyrillic.

I went through all the existing mods and found these at least to make trees. Maybe we can get some inspiration from here. We could integrate it into FoodIndustrie or make a depency to it.

https://mods.factorio.com/mod/Bio_Industries
https://mods.factorio.com/mod/Treefarm-Lite-fix
https://mods.factorio.com/mod/Arborium
https://mods.factorio.com/mod/WoodHarvester

And these for the cattle:

https://mods.factorio.com/mod/BiterPets

I would be very happy if you could do the art. It's something I'll never learn.

Oceanel51 committed:
trees cultivation

Added trees cultivation
balance and fixes

Oceanel51 commented:
Here's a quick tree made.

This is not a ready option. Todo:
1. Deal with autoplace:
- I want to get at the start of 10-20 trees in sight
- further on the map you need to meet whole forests of 500-1000 trees. But, they can also be grown on farms
2. Several options for tree sprites. Big, smaller, turned.

They will need a lot for the production of juices.

Commit:
added oranges and a foodpicker. It's looks like the fisher robot. But put it next to a tree and it will start picking fruit.

Commit:
Merge branch 'master' of https://github.com/jonkeda/Food-Industry

# Conflicts:
#	graphics/icons/items/orange.png

jonkeda commented:
I like the trees a lot. 

I had a look at the other Tree mods. And the bio industries one looked the best. It allows for free form creation of forests of trees. And that combined with the Food picker could already be fun. And I could probably add flyingbots that pick fruit for larger areas.

So fruit would be => Wild orange tree => Orange => Seeds => Sapling => Planted orange tree. And same idea for apples.


https://mods.factorio.com/mod/Bio_Industries

Oceanel51 commented:
> I had a look at the other Tree mods. And the bio industries one looked the best.
> 
The Bio_Industries is one of the my favorite mods!

> I like the trees a lot.
> So fruit would be => Wild orange tree => Orange => Seeds => Sapling => Planted orange tree. And same idea for apples.
> 
Yes, you catch the meaning!

In my last [commit](https://github.com/Oceanel51/Food-Industry/pull/4/commits/2e089ddb917feeed0f30c0bf0678dca05f566105) trees work like plants. Due to the fact that fruit trees bear fruit for about 10 years, then they are transplanted to new ones. Therefore, it is meaningless to destroy trees after receiving one crop.
In the game mechanics of factorio we could do this, Conception 2:
- something planted seedlings according to the planer, as well as the ability to plant a sapling in Bio_Industries, but maybe it will be a crafting-mashine entity with a tree texture!
- then for some time it grows, let it be 20-30 minutes
- then begins to bear fruit, for example, every 5-10 minutes, 5-40 fruits (depending on the type of tree)
- additionally collect plants mesh and some raw wood
- the tree is gradually depleted (dies)
- after a certain amount of harvest - only raw wood can be obtained from the tree
- the tree collapses and it requires a new planting of seedlings
- cycle repeats.
At the same time collecting can be carried out by incerters, maybe by robots.

This concept is even better than Conception 1 from https://trello.com/c/G4wF2X5q, just invented :)
It's all difficult, but doable.

Oceanel51 commented:
have you seen the mod https://mods.factorio.com/mod/Bioluminescence? That's wonderful!
![2019-02-05_152957](https://user-images.githubusercontent.com/19427757/52408935-b6d38d00-2adc-11e9-85ce-b55985e293e4.png)

I would like to see fruit tree gardens on the map!

Oceanel51 commented:
I think it makes sense to even make several options for growing trees. Only need to balance the numbers. People and moods are different, someone wants to build a long and profitable production chain, someone is cheaper and faster.

jonkeda commented:
I guess it does. I started with the simplest case first.

Wild orange tree (cut down) => Orange (Process) => Seeds (Grow in house) => Sapling (plant on ground) => Planted orange tree. 

Could you push it through once that is done? I really need to apples and oranges for the colonist and would like to push that mod also to get feedback from people.

Oceanel51 committed:
tree fixes

Commit:
Merge branch 'master' of https://github.com/jonkeda/Food-Industry

# Conflicts:
#	graphics/icons/items/orange.png

Commit:
Merge branch 'master' of https://github.com/jonkeda/Food-Industry

Commit:
Added a Tree_Greenhouse

Moved apples and oranges definitions to Trees.lua

Currently:

Wilde--tree => Fruits => Seeds => Grow seedlings in tree_greenhouse => plant seedlings directly as trees

Yet another step closer to were we want to get.

Commit:
Trees now grow from seedlings. Takes 10 seconds for a seedling to become a tree

Currently:

Wilde--tree => Fruits => Seeds => Grow seedlings in tree_greenhouse => plant seedlings => grow in to tree

Fruittrees also have a maximum number of fruits that can be picked, currently 100

Commit:
Added a Fruit-Scissor. This can cut apples and oranges from trees.

Disabled the recipes for the fruit and fish inserters and added them to the Automation and Logistics technology.

Changed the icons of the foodpicker and changed the platforms.

Commit:
Added a Fruit-Scissor. This can cut apples and oranges from trees.

Disabled the recipes for the fruit and fish inserters and added them to the Automation and Logistics technology.

Changed the icons of the foodpicker and changed the platforms.

Commit:
Added cattle.

Cattle spawner => cattle-calfs => cattle-feeder + straw => Cattle => cattle-grabber => cattle-butcher => biter-meat

Changed location of fishing and tree machinery to its own rows.

Recipe, ingredient, result and pre-requisites balancing needed.

Commit:
Copied the poorman-wood-fence from bioindustries and created a very weak ( 1 health ) wooden fence. Meant to keep the cattle in.

Also copied the gate. But the graphics need to be changed from concrete to wood. Help?

Also. The cattle feeder doesn't work yet.

Commit:
Fixed some issues with the inserts.

And they are updating their positions now. Looks much better.

Commit:
Merged with version 2.0

Commit:
Version 2.0

jonkeda commented:
Closed it and merge to Version 2.0 branche including V2 changes
